packages/client-admin/src/context/auth.context.tsx

```diff
@@ -17,21 +19,32 @@ export interface AuthProviderProps {
 
 export const AuthProvider: FC<AuthProviderProps> = (props) => {
   const [token, setToken] = useState<Token>(restoreToken());
+  const [decoded_token, setDecodedToken] = useState<any>(restoreDecodedToken());
 
   useEffect(() => {
     saveToken(token);
+    saveDecodedToken(jwt_decode(token.token));
   }, [token]);
 
-  return <AuthContext.Provider value={{ token, setToken }} {...props} />;
+  return <AuthContext.Provider value={{ token, decoded_token, setToken }} {...props} />;
 };
 
 const saveToken = (token: Token) => {
   localStorage.setItem('token', JSON.stringify(token));
 };
 
+const saveDecodedToken = (decoded_token: any) => {
```

Comment:
You dont need to save or restore the decoded token if you are saving the token itself.

Every time the token changes just decode it.

That way there isn't a conflict between a token and a decoded token